git-glossary.md

```diff
@@ -0,0 +1,50 @@
+# Git Glossary 
+
+This git glossary contains definitions of terms used in this repository.
+
+There is another dcoument that describes a mental model of git that
+when understood can help one master git concepts at a deeper level. 
+
+## commit
+As a noun: A single point in history of a repository. The entire history of a project is represented as a set
+of interrelated commits. Each commit is uniquely identified by a 40-character hexadecimal string which is created using
+a hash function, also referred to has the SHA of the commit. Sometimes the SHA is abbreviated to the 
+first 7 characters of the 40-character string.
+
+As a verb: The action of storing a new snapshot of the project's state in the Git history, by creating a new commit. 
+
+## commit object
+An object which contains the information about a particular commit.
+
+## git
+`git` is a modern version control system used to maintain a history of changes to source code for a project. `git` has a 
+distributed architecture and is an examnple of a DVCS (Distrubuted Version Control System). Rather than have only a single
+place where the full version historyof the software is stored (like CVS or Subversion), in `git` every developer's working copy
+of the code is also a repository that can contain the full history of all changes.
+
+## github
+`github` is a website and a service that maintains git repositories for project teams. Developers can interact with repositories
+atored in `gtihub` using it's web interface or using the `git` command line interface.
+
+## github account
+You must have an account to make changes to repositories stored on `github`. There are two types of github accounts, organization
+and individual. The organization account is often where the main repository for open source projects resides. The person (or persons)
+with access to the organization account control who can make changes to the repositories owned by the organization account.
```

Comment:
Can we add that this is called "push access"? (Though in a way this may be confusing as the workflow we are discussing elsewhere doesn't use push to the repo)